csfieldguide/chapters/content/en/coding-compression/sections/general-purpose.md

```diff
@@ -43,27 +23,308 @@ Here are some of the bits from the example earlier in this chapter; you can past
+| 1    | 000000000001                 | A        |
+| 2    | 000000000010                 | M        |
+| 3    | 000000000011                 | S        |
+| 4    | 000000000100                 | IA       |
+
+We then proceed to the next character in the message, adding a new sequence to the dictionary each with each step:
+
+| Step | Current Sequence | Next Character | Code | New Dict Entry Code | New Dict Sequence |
+|------|------------------|----------------|------|---------------------|-------------------|
+| 2    | A                | M              | 1    | 5                   | AM                |
+| 3    | M                | S              | 2    | 6                   | MS                |
+| 4    | S                | A              | 3    | 7                   | SA                |
+| 5    | AM               | S              | 5    | 8                   | AMS               |
+
+In step 5, the sequence is `A`, and the next character is `M`. But the sequence `AM` is already in the dictionary, so we replace the whole sequence with the corresponding dictionary code (`5`), and add `AMS` to the dictionary instead (as `S` is the next character).
+
+| Step | Current Sequence | Next Character | Code | New Dict Entry Code | New Dict Sequence |
+|------|------------------|----------------|------|---------------------|-------------------|
+| 6    | SA               | M              | 7    | 9                   | SAM               |
+| 7    | M                | I              | 2    | 10                  | MI                |
+| 8    | IA               | M              | 4    | 11                  | IAM               |
+| 9    | M                | -              | 2    | -                   | -                 |
+
+When we have finished encoding the string, we now have a dictionary that looks like this:
+
+| Code | Code (Binary Representation) | Sequence |
+|------|------------------------------|----------|
+| 0    | 000000000000                 | I        |
+| 1    | 000000000001                 | A        |
+| 2    | 000000000010                 | M        |
+| 3    | 000000000011                 | S        |
+| 4    | 000000000100                 | IA       |
+| 5    | 000000000101                 | AM       |
+| 6    | 000000000110                 | MS       |
+| 7    | 000000000111                 | SA       |
+| 8    | 000000001000                 | AMS      |
+| 9    | 000000001001                 | SAM      |
+| 10   | 000000001010                 | MI       |
+| 11   | 000000001011                 | IAM      |
+
+And the string has been replaced with a series of codes, each corresponding to a value in the dictionary.
+`0 1 2 3 5 7 2 4 2`
+
+To see just how much compression we achieved, let’s take a look at the binary behind the characters. We'll assume that the characters are coded in 8 bits each (this is typical), based on UTF-8 coding (which is the same as ASCII for English characters). In binary, originally our message looked like this:
```

Comment:
I think this will confuse students, leaving it as is.

Reply:
The misinformation is minor but I still don't like calling non-English stuff like this as wholly English because this will be translated into languages that use the same character set (Maori, most Germanic languages etc) - there are people who will read this content in their native language and feel insulted by the appropriation.
Instead it could say `(which is the same as ASCII for these characters)` or something to that extent and avoid languages altogether